examples/tutorials/Part 07 - Federated Learning with Federated Dataset.ipynb

```diff
@@ -106,7 +106,10 @@
     "\n",
     "# Build the FederatedDataset object\n",
     "dataset = sy.FederatedDataset(datasets)\n",
-    "print(dataset.workers)"
+    "print(dataset.workers)\n",
+    "optimizers = {}\n",
+    "for worker in dataset.workers:\n",
+    "    optimizers[worker] = th.optim.Adam(params=model.parameters(),lr=1e-2)"
```

Comment:
Why do you have to do this?
You're giving the same references to model parameters to all optimizers so the result should be the same, or not? (I don't know how Adam works)

Reply:
I don't remember exactly the reason, but currently in PySyft having one joint optimizer across locations doesn't work with optimizers that use momentum (including Adam). 
And I switched to Adam as searching on internet I noticed that other people had problems getting the training to converge with SGD as well.

Reply:
Oh yes you're right, the pb is with momentum!